WalletWasabi/Nostr/NostrCoordinatorPublisher.cs

```diff
@@ -0,0 +1,49 @@
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+using NBitcoin.Secp256k1;
+using NNostr.Client;
+using WalletWasabi.Bases;
+using WalletWasabi.Logging;
+
+namespace WalletWasabi.Nostr;
+
+public class NostrCoordinatorPublisher : PeriodicRunner
+{
+	private readonly Uri[] _relayUris = [new("wss://relay.primal.net")];
+
+	public NostrCoordinatorPublisher(TimeSpan period, ECPrivKey key, NostrCoordinator coordinator) : base(period)
+	{
+		Coordinator = coordinator;
+		Key = key;
+		Client = NostrExtensions.Create(_relayUris, (EndPoint?)null);
+	}
+
+	private INostrClient Client { get; }
+
+	private NostrCoordinator Coordinator { get; }
+
+	private ECPrivKey Key { get; }
+
+	protected override async Task ActionAsync(CancellationToken cancel)
+	{
+		try
+		{
+			var discoveryEvent = await Key.CreateCoordinatorDiscoveryEventAsync(Coordinator).ConfigureAwait(false);
+
+			using var cts = new CancellationTokenSource(TimeSpan.FromMinutes(3));
```

Comment:
3 minutes! No no, it could be 5 or 10 seconds maximum because if after 5/10 seconds the event was not published then something went really wrong.